test/api.test.js

```diff
@@ -0,0 +1,48 @@
+// test src/background/api
+
+import { fail } from 'assert';
+
+import * as api from '../src/background/api'
+import * as request from "../src/background/api/request"
+import rpc from "../src/background/api/rpc"
+
+// ./node_modules/.bin/jest -t 'make get account fail'
+test('make get account fail', async () => {
+  // mock commonFetch to throw
+  const spy = jest.spyOn(request, 'commonFetch');
+  // spy.mockReturnValue({foo: 'bar'});
+  spy.mockImplementation(() => {
+    throw new Error();
+  });
+  let tmp
+  try { 
+    tmp = await api.getBalance('blah')  
+    fail('it should have thrown')
+  } catch (e) {
+    // passthrow
+  }
+  expect(tmp).toEqual({});
+  spy.mockRestore();
+})
+
+// ./node_modules/.bin/jest -t 'get account should fallback to RPC on failure'
+test('get account should fallback to RPC on failure', async () => {
+  const mockGetRpcAccount = jest.spyOn(rpc, 'getRpcAccount');
+  mockGetRpcAccount.mockReturnValue({data: {foo: 'bar'}});
+  // mock commonFetch to throw
+  const mockCommonFetch = jest.spyOn(request, 'commonFetch');
+  mockCommonFetch.mockImplementation(() => {
+    throw new Error();
+  });
+  let tmp
+  try { 
+    tmp = await api.getBalance('blah')  
+    fail('it should have thrown')
+  } catch (e) {
+    // passthrow
+  }
```

Comment:
I don't understand this code. It looks exactly equal to
```suggestion
  let tmp = await api.getBalance('blah')  
```

`getBalance` doesn't throw. `fail` throws. And gets caught and continues.